src/khan.ts

```diff
@@ -0,0 +1,133 @@
+import net from 'net'
+import { Cell, jam, dwim, Noun, enjs } from '@urbit/nockjs'
+import { newtDecode, newtEncode } from './bytes'
+
+// returns a js Noun Object with the structure of a %fyrd
+function Fyrd(desk: string, thread: string, mark_out: string, mark_in: string, atom: string) {
+  return dwim([
+    0,
+    'fyrd',
+    desk,
+    thread,
+    mark_out,
+    mark_in,
+    atom
+  ]);
+}
+
+// returns a js Fyrd, whose trailing atom is
+// some hoon code to be run as a thread
+function Thread (hoon: string) {
+  return Fyrd('base', 'khan-eval', 'noun', 'ted-eval', hoon);
+}
+
+interface ThreadResponse {
+  mark: string,
+  node: {
+    [key: string]: {
+      mark: string;
+      node: Noun;
+    }
+  }
+}
+
+function unwrapUnit(noun: Noun): Noun | null {
+  if (noun instanceof Cell && noun.head.equals(dwim(0))) {
+    return noun.tail;
+  }
+
+  return null;
+}
+
+// decodes response and unpacks it to a noun,
+// if it has the expected structure of a fyrd response
```

Comment:
does this handle arbitrarily large trees and convert them to a JSON tree of JS primitives? How? is enjs documented somewhere other than the code?